master/buildbot/process/buildrequestdistributor.py

```diff
@@ -504,12 +504,12 @@ def _maybeStartBuildsOnBuilder(self, bldr, _reactor=reactor):
                 bc = self.createBuildChooser(bldr, self.master)
                 continue
 
-            buildStarted = yield bldr.maybeStartBuild(worker, breqs)
-            if not buildStarted:
-                yield self.master.data.updates.unclaimBuildRequests(brids)
-                # try starting builds again.  If we still have a working worker,
-                # then this may re-claim the same buildrequests
-                self.botmaster.maybeStartBuildsForBuilder(self.name)
+            d = bldr.maybeStartBuild(worker, breqs)
+
+            @d.addCallback
+            def isItStarted(buildStarted):
+                if not buildStarted:
+                    yield self.master.data.updates.unclaimBuildRequests(brids)
```

Comment:
Hm, this is wrong, since this function is using `inlineCallbacks`.  See http://docs.buildbot.net/latest/developer/style.html#inlinecallbacks and various other Twisted tutorial for more information.

I'm not sure what you're trying to do with this change -- it looks like you've just taken a correct use of inlineCallbacks and replaced it with an incorrect one (which would lead logically to test failures or worse)